laurent22/joplin
# Mobile: Resolves #10763: Support permanent note deletion on mobile

# Summary

This pull request **partially** fixes #10763 by adding a "Permanently delete note" option on Android.

To limit code duplication, it refactors the mobile deletion UI to use the existing `deleteNote` and `permanentlyDeleteNote` commands.

# Changes

- Moves the `deleteNote` and `permanentlyDeleteNote` commands to `packages/lib/commands/` to allow use on mobile.
- Refactors `permanentlyDeleteNote` to use `shim.showMessageBox`, rather than Electron-specific logic.
- Updates the mobile reducer logic in [root.tsx](https://github.com/laurent22/joplin/compare/dev...personalizedrefrigerator:pr/mobile/permanently-delete-note-on-android-2?expand=1#diff-b30ec16beba186d84c76e720993c1b30c83e2ede06fe0248555a3582d95587e7) to navigate away from the selected note on deletion.

# Testing plan

**Manual testing**:
1. Start Joplin
2. Open/create a folder with at least 3 notes.
3. From the "notes" screen, delete 1-2 notes.
4. Verify that the notebook list is shown after deleting each.
5. Open the trash folder
6. Open one of the notes.
7. Permanently delete it from the note actions menu.
8. Verify that the list of notes in the trash folder is shown.
9. Open the other note.
10. Permanently delete it.
11. Verify that both notes are now missing from the trash folder.
12. Delete another note to the trash folder.
13. Restore the note.

This has been tested successfully on an Android (API 35) emulator.
